### PR TITLE
refactor: boot user's settings

### PR DESCRIPTION
**Pull request by sshanzel**
DD-337

**Relevant API that publishes the message**: https://github.com/dailydotdev/daily-api/pull/834

The subscription end for utilizing the boot query with the user's settings.

The default values were fetched from the info in our APIs entity.

I have created a generic subscription handler to accommodate cases where we just need to store the whole object and don't require any post-processing of the data.

I have also refactored the fetching of `alerts` related data since I have created a utility function that accepts parameters for the same cases. The mentioned function is now being utilized for fetching both `alerts` and `settings`.

~~Initially, I have tried to create a generic class to accommodate these kinds of cases since they almost look like the same from alerts POV. Unfortunately, things will become tightly coupled and it will be hard to make some changes in the future.~~

~~So I decided to create one on its own.~~

~~The PR also includes a stub value for~~
- ~~`appInsaneMode`: `true` - from API DB it defaults to true.~~
- ~~`enableCardAnimations`: `true` - from API DB it defaults to true.~~

### CODE REVIEW

**Review by idoshamun — CHANGES_REQUESTED** [review]
This looks neat. I love the fact that it becomes easier to embed more properties to the boot query with time.
I think we can remove the legacy properties from the defaults object because a client which relies on the boot query no longer expect to use these.
I do think that we should try structure this process. The alerts and settings flow look pretty much the same. You can create a function that receives some other functions as helpers. But the general workflow is the same so I recommend structuring it into a function.

File: src/redis.js [hunk 1]
```diff
@@ -27,9 +31,18 @@ export function deleteKeysByPattern(pattern) {
...
+export const SETTINGS_DEFAULT = {
+  theme: 'darcula',
+  enableCardAnimations: true,
+  showTopSites: true,
+  insaneMode: false,
+  appInsaneMode: true,
+  spaciness: 'eco',
+  showOnlyUnreadPosts: false,
+  openNewTab: true,
+  openSidebar: true,
```
Comment (rebelchris) [inline src/redis.js:44]: Do note this is now: `sidebarExpanded`

File: src/redis.js [hunk 2]
```diff
@@ -51,20 +64,46 @@ export const getAlertsFromAPI = async (ctx) => {
...
 
+export const getSettingsFromAPI = async (ctx) => {
+  const query = `{
+    userSettings {
+      openNewTab
+      showOnlyUnreadPosts
+      theme
+      spaciness
+      insaneMode
+      showTopSites
```
Comment (rebelchris) [inline src/redis.js:75]: I think `sidebarExpanded` is already available to be fetched here.

**Review by rebelchris — CHANGES_REQUESTED** [review]
Wow this is looking pretty slick man!
Really nice we can quickly serve this to prevent further delay on the basic settings.

File: src/redis.js (hunk 1 above)
Comment (sshanzel) [reply src/redis.js:44]: Ah, we were just literally discussing this earlier 😆  thank you for the mention! 🚢

**Comment by sshanzel** [comment]
Once the [relevant API ](https://github.com/dailydotdev/daily-api/pull/834)for this integration has been merged, we can then safely review this PR.

**Comment by sshanzel** [comment]
This should now be ready for review since the relevant API has been merged 🚢

**Review by idoshamun — APPROVED** [review]
Wow! This PR is so great.
I love how you abstracted away the workflow of using the cache and introducing something new to the boot query

**Comment by sshanzel** [comment]
@idoshamun thank you for the awesome review!

@rebelchris kindly let me know if you require further changes. Thank you! 🚢

File: src/redis.js [hunk 3]
```diff
@@ -27,52 +30,38 @@ export function deleteKeysByPattern(pattern) {
...
-      const alerts = res.data.userAlerts;
+      const data = await getFromApi();
 
-      await setRedisAlerts(alertsKey, alerts);
+      await setRedisObject(userId, data);
 
-      return alerts;
+      return data;
     } catch (ex) {
       // TODO: use a dedicated logger for exceptions
```
Comment (rebelchris) [inline src/redis.js:60]: Should we do anything with this at this point?
Maybe even a console.error so they at least show up?

**Review by rebelchris — APPROVED** [review]
Wow really nice work here @sshanzel 
Super cool what you've done with this.

Just one small non-blocking comment.

File: src/redis.js (hunk 3 above)
Comment (sshanzel) [reply src/redis.js:60]: Very good point! Let's use a console.error for this to make debugging much easier.

**Comment by sshanzel** [comment]
Thank you both! 🚢

**Pull request merged**